keyvault/client/src/main/java/com/azure/keyvault/SecretClient.java

```diff
@@ -0,0 +1,426 @@
+     *
+     * <p>It is possible to get full Secrets with values from this information. Loop over the {@link SecretBase secretBase} and
+     * call {@link SecretClient#getSecret(SecretBase)} . This will return the {@link Secret} secrets with values included of its latest version.</p>
+     * <pre>
+     * List&lt;Secret&gt; secrets = new ArrayList();
+     * secretClient.listSecrets().stream().map(secretClient::getSecret).forEach(secretResponse -&gt;
+     *   secrets.add(secretResponse.value()));
+     * </pre>
+     *
+     * @return A {@link List} containing {@link SecretBase} of all the secrets in the vault. The {@link SecretBase} contains all the information about the secret, except its value.
+     */
+    public List<SecretBase> listSecrets() {
+        return service.getSecrets(vaultEndpoint, DEFAULT_MAX_PAGE_RESULTS, API_VERSION, ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE)
+            .flatMapMany(this::extractAndFetchSecrets).collectList().block();
+    }
+
+    /**
+     * Lists {@link DeletedSecret deleted secrets} of the key vault. The get deleted secrets operation returns the secrets that
+     * have been deleted for a vault enabled for soft-delete. This operation requires the {@code secrets/list} permission.
+     *
+     * <p><strong>Code Samples</strong></p>
+     * <pre>
+     * secretClient.listDeletedSecrets().stream().forEach(deletedSecret ->
+     *   System.out.println(String.format("Deleted secret's recovery Id %s", deletedSecret.recoveryId())));
+     * </pre>
+     *
+     *
+     * @return A {@link List} containing all of the {@link DeletedSecret deleted secrets} in the vault.
+     */
+    public List<DeletedSecret> listDeletedSecrets() {
+        return service.getDeletedSecrets(vaultEndpoint, DEFAULT_MAX_PAGE_RESULTS, API_VERSION, ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE)
+            .flatMapMany(this::extractAndFetchDeletedSecrets).collectList().block();
+    }
+
+    /**
+     * List all versions of the specified secret. The individual secret response in the list is represented by {@link SecretBase}
+     * as only the base secret identifier and its attributes are provided in the response. The secret values are
+     * not provided in the response. This operation requires the {@code secrets/list} permission.
+     *
+     * <p>It is possible to get full Secrets with values for each version from this information. Loop over the {@link SecretBase secretBase} and
+     * call {@link SecretClient#getSecret(SecretBase)} . This will return the {@link Secret} secrets with values included of the specified versions.</p>
+     * <pre>
+     * List&lt;Secret&gt; secretVersions = new ArrayList();
+     * secretClient.listSecretVersions("secretName").stream().map(secretClient::getSecret).forEach(secretResponse -&gt;
+     *   secretVersions.add(secretResponse.value()));
+     * </pre>
```

Comment:
Not a fan of this code style - either do a .collect after the .map, or even better, just do something in the forEach that doesn't add it into a List.

Reply:
updated